src/main.cpp

```diff
@@ -2558,21 +2571,43 @@ bool Session::InitializeDRM()
 
     unsigned char key_system[16];
     AP4_ParseHex(strkey.c_str(), key_system, 16);
+    uint32_t currentSessionTypes = 0;
 
     for (size_t ses(1); ses < cdm_sessions_.size(); ++ses)
     {
       AP4_DataBuffer init_data;
       const char* optionalKeyParameter(nullptr);
+      adaptive::AdaptiveTree::Period::PSSH sessionPsshset =
+        adaptiveTree_->current_period_->psshSets_[ses];
+      uint32_t sessionType = 0;
```

Comment:
IMO this is not very clear, here we should make it easier understand that we handle flag operations,
at least we should change all these integer variables to handle a enum, something like this:
```c++
enum MediaTypeFlags
{
MEDIA_TYPE_UNDEFINED = 0,
MEDIA_TYPE_VIDEO,
MEDIA_TYPE_AUDIO
...
```
if it is too much work for this PR, keep it for later cleanup

Reply:
Thanks for pointing out however let's keep for a later cleanup